db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2392,10 +2403,21 @@ void DBImpl::DisableManualCompaction() {
   }
 }
 
+// NOTE: Calling EnableManualCompaction() may overwrite the
+// user-provided canceled variable in CompactRangeOptions
 void DBImpl::EnableManualCompaction() {
   InstrumentedMutexLock l(&mutex_);
   assert(manual_compaction_paused_ > 0);
   manual_compaction_paused_.fetch_sub(1, std::memory_order_release);
+
+  // Reset canceled as false when this is the last time call
+  // Mark the canceled as false to enable manual compaction (may
+  // overwrite user-provided `canceled`)
+  if (manual_compaction_paused_ == 0) {
+    for (const auto& manual_compaction : manual_compaction_dequeue_) {
+      manual_compaction->canceled = false;
+    }
+  }
```

Comment:
Oh I didn't realize this was still here. Let me rethink it.

Reply:
I still don't think we need to support the case of uncanceling manual compactions that did not detect the cancellation between `DisableManualCompaction()` and `EnableManualCompaction()`. @jay-zhuang added support to kick out queued manual compactions when the disable happens, so those will never be uncanceled. And for running manual compactions, the `CompactionIterator` is checking the canceled atomic on every iteration, so there might be uncanceled manual compactions only when the gap between disable/enable is incredibly (unbelievably?) short.

Since it isn't obvious why this is needed, and it introduces API complexity including wrongly uncanceling in certain cases, I think we should justify its necessity or remove it.

Reply:
yeah, that's true, we don't need to handle the manual compactions in the dequeue. Thanks.